### PR TITLE
Set ACPI RSDP location and kernel command line in zero_page

### PR DESCRIPTION
**Pull request by andrisaar**
Some low-level changes for the future Linux kernel compatibility:

 - set the ACPI RSDP location in the zero page structures. The Linux kernel won't search EBDA for RSDP, it expects the value in the zero page to point to the correct location.
 - if there's a kernel command line available in fw_cfg, load that into memory and pass it to the kernel as well.

### CODE REVIEW

File: stage0/src/main.rs [hunk 1]
```diff
@@ -259,6 +259,23 @@ pub extern "C" fn rust64_start(encrypted: u64) -> ! {
         zero_page.hdr.setup_data = &setup_data.header as *const oak_linux_boot_params::SetupData;
     }
 
+    if let Ok(cmdline_size) = fwcfg.read_cmdline_size() {
+        if cmdline_size > 0 {
+            let mut buf = alloc
+                .alloc(Layout::from_size_align(cmdline_size as usize, 1).unwrap())
```
Comment (mariaschett) [inline stage0/src/main.rs:265]: Do we want to change this `unwrap` to expect too?

File: stage0/src/acpi.rs [hunk 2]
```diff
@@ -384,7 +384,7 @@ impl RomfileCommand {
     }
 }
 
-pub fn build_acpi_tables(fwcfg: &mut FwCfg) -> Result<(), &'static str> {
+pub fn build_acpi_tables(fwcfg: &mut FwCfg) -> Result<u64, &'static str> {
```
Comment (tiziano88) [inline stage0/src/acpi.rs:387]: Could you add a comment explaining what is being returned?

**Review by tiziano88 — APPROVED** [review]

File: stage0/src/fw_cfg.rs [hunk 3]
```diff
@@ -24,9 +24,23 @@ const FWCFG_PORT_DATA: u16 = 0x511;
 
 const SIGNATURE: &[u8] = b"QEMU";
 
+/// Selector keys for "well-known" fw_cfg entries.
+///
+/// See QEMU include/standard-headers/linux/qemu_fw_cfg.h for the authoritative list.
+#[allow(dead_code)]
```
Comment (mariaschett) [inline stage0/src/fw_cfg.rs:30]: Why did you add more than the one you need now?

**Review by mariaschett — APPROVED** [review]
LGTM.

File: stage0/src/main.rs (hunk 1 above)
Comment (andrisaar) [reply stage0/src/main.rs:265]: I don't think this can actually fail, because (a) cmdline_size is an `u32` and therefore by always smaller than `isize::MAX`, and we fix the alignment itself to be 2^0 = 1. But I've changed it to an `expect()`, just in case.

File: stage0/src/acpi.rs (hunk 2 above)
Comment (andrisaar) [reply stage0/src/acpi.rs:387]: Done.
